src/Microsoft.NetCore.Analyzers/Core/Composition/DoNotMixAttributesFromDifferentVersionsOfMEF.cs

```diff
@@ -78,12 +78,16 @@ private static void AnalyzeSymbol(SymbolAnalysisContext symbolContext, IEnumerab
             }
 
             var badNamespaces = exportAttributes.Except(appliedExportAttributes).Select(s => s.ContainingNamespace).ToSet();
+            var goodNamespaces = appliedExportAttributes.Select(s => s.ContainingNamespace).ToSet();
 
-            // Now look at all attributes and see if any are metadata attributes
+            // Now look at all attributes and see if any are metadata attributes from badNamespaces, but none from good namepaces.
             foreach (var namedTypeAttribute in namedTypeAttributes)
             {
-                if (namedTypeAttribute.AttributeClass.GetApplicableAttributes().Any(ad => badNamespaces.Contains(ad.AttributeClass.ContainingNamespace) &&
-                                                                                                          ad.AttributeClass.Name == "MetadataAttributeAttribute"))
+                var appliedAttributes = namedTypeAttribute.AttributeClass.GetApplicableAttributes();
+                if (appliedAttributes.Any(ad => badNamespaces.Contains(ad.AttributeClass.ContainingNamespace) &&
+                        ad.AttributeClass.Name == "MetadataAttributeAttribute") &&
+                    !appliedAttributes.Any(ad => goodNamespaces.Contains(ad.AttributeClass.ContainingNamespace) &&
+                        ad.AttributeClass.Name == "MetadataAttributeAttribute"))
```

Comment:
Can we just filter these down to all `ad.AttributeClass.Name == "MetadataAttributeAttribute"` with Linq? It'll make this statement a lot more concise, since we need that filter in both `Any` statements